planner/core/preprocess.go

```diff
@@ -833,12 +833,27 @@ func (p *preprocessor) checkDropSequenceGrammar(stmt *ast.DropSequenceStmt) {
 }
 
 func (p *preprocessor) checkDropTableGrammar(stmt *ast.DropTableStmt) {
-	p.checkDropTableNames(stmt.Tables)
+	currentDB := model.NewCIStr(p.ctx.GetSessionVars().CurrentDB)
 	enableNoopFuncs := p.ctx.GetSessionVars().EnableNoopFuncs
-	if stmt.TemporaryKeyword == ast.TemporaryLocal && !enableNoopFuncs {
-		p.err = expression.ErrFunctionsNoopImpl.GenWithStackByArgs("DROP TEMPORARY TABLE")
-		return
+	for _, t := range stmt.Tables {
```

Comment:
Do this only when `stmt.TemporaryKeyword != ast.TemporaryNone`, so that it doesn't affect the original logic.